src/app/questionary/containers/questionary-review-page/questionary-review-page.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Question, QuestionaryAnswer } from '../../models/questionary-models';
+import { QuestionaryService } from '../../services/questionary.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { tap } from 'rxjs/operators';
+import { noop } from 'rxjs';
+import { UrlService } from '../../../services/url.service';
+
+@Component({
+  selector: 'questionary-review-page',
+  templateUrl: './questionary-review-page.component.html',
+  styleUrls: ['./questionary-review-page.component.scss']
+})
+export class QuestionaryReviewPageComponent implements OnInit, OnDestroy {
+  questionaryId: string;
+  loadingQuestionary: boolean;
+  questionary: QuestionaryAnswer;
+  questions: Question[];
+  unsubscribe = [];
+
+  constructor(
+    private questionaryService: QuestionaryService,
+    private route: ActivatedRoute,
+    private urlService: UrlService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.loadingQuestionary = true;
+    this.questionaryId = this.route.snapshot.paramMap.get('questionaryId');
+    console.log(this.questionaryId);
+    const questionaryService = this.questionaryService.fetchQuestionary(this.questionaryId).pipe(
+      tap((questionary) => {
+        this.questionaryService.setQuestionary(questionary);
+        this.questionary = questionary;
+        this.questions = questionary.questions;
```

Comment:
Porque tem mais uma variável questions se ela já ta dentro da variável questionary?
De duas uma:
1) Nem sempre vai ter questions dentro de questionary: aí você tem que fazer verificação.
`this.questionary = questionary
if (this.questionary && this.questionary.questions) {
    this.questions = this.questionary.questions
}`

2) Sempre vai ter. Então ao invés de usar this.questions, pode usar this.questionary.questions

Reply:
Ficou péssima a formatação do código, sry